### ISSUE TITLE
Path Parameter Definitions Not Required Across Operations

### ISSUE THREAD

**Issue by dillonredding** [stoplightio/spectral#1076]
**Describe the bug**
The `path-params` rule is not triggered when a path parameter is defined under one operation but not in another, both of which are under the same path item.

**To Reproduce**
Given the following OpenAPI document:

```yaml
openapi: 3.0.3
info:
  title: API Title
  version: '1.0'
paths:
  /test/{foo}:
    get:
      parameters:
        - name: foo
          in: path
          required: true
          schema:
            type: string
      responses:
        '200':
          description: OK
    put:
      responses:
        '200':
          description: OK
```

I get this when linting:

```
$ spectral lint .\openapi.yaml
OpenAPI 3.x detected

c:/dev/openapi.yaml
  1:1  warning  oas3-api-servers       OpenAPI `servers` must be present and non-empty array.
  1:1  warning  openapi-tags           OpenAPI object should have non-empty `tags` array.
  2:6  warning  info-contact           Info object should contain `contact` object.
  2:6  warning  info-description       OpenAPI object info `description` must be present and non-empty string.
  7:9  warning  operation-description  Operation `description` must be present and non-empty string.
  7:9  warning  operation-operationId  Operation should have an `operationId`.
  7:9  warning  operation-tags         Operation should have non-empty `tags` array.
 17:9  warning  operation-description  Operation `description` must be present and non-empty string.
 17:9  warning  operation-operationId  Operation should have an `operationId`.
 17:9  warning  operation-tags         Operation should have non-empty `tags` array.

✖ 10 problems (0 errors, 10 warnings, 0 infos, 0 hints)
```

**Expected behavior**
This should have triggered the `path-params` rule for the `put`, which is what I get without the parameter definition in the `get`.

```
6:15    error  path-params            The path `/test/{foo}` uses a parameter `{foo}` that does not have a corresponding definition.
```

**Environment:**
 - Library version: 5.3.0
 - OS: Windows 10

**Comment by nulltoken** [comment]
@dillonredding Good catch! On it.

**Comment by nulltoken** [comment]
@dillonredding #1077 should fix that.

**Comment by dillonredding** [comment]
You're awesome! Thank you so much!
